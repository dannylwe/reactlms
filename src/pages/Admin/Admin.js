import React, { Component } from "react";
import axios from "axios";
import "../Dashboard/Dashboard.css";

class Admin extends Component {
  state = {
    items: ""
  };
  componentDidMount() {
    axios
      .get("https://challenge3andela.herokuapp.com/api/v1/parcels/all", {
        withCredentials: true
      })
      .then(res => {
        this.setState({ items: res.data["item info"] });
      });
  }

  render() {
    const { items } = this.state;

    const itemValues =
      items.length > 0
        ? items.map(item => (
            <tr>
              <th>{item.destination}</th>
              <th>{item.nickname}</th>
              <th>{item.status}</th>
              <th>{item.pickup}</th>
            </tr>
          ))
        : "loading......";
    return (
      <>
        <nav class="navbar navbar-dark bg-dark">
          <a class="navbar-brand" href="/dashboard">
            Home
          </a>
          <a class="navbar-brand" href="/login">
            Logout
          </a>
        </nav>

        <br />
        <table>
          <tr>
            <th>Destination</th>
            <th>Order Nickname</th>
            <th>Status</th>
            <th>Pickup Location</th>
          </tr>
          {itemValues}
        </table>
      </>
    );
  }
}

export default Admin;
