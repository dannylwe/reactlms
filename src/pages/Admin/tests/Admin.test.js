import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import Admin from "../Admin";

describe("Admin", () => {
  it("should render without failing", () => {
    const wrapper = renderer.create(
      <MemoryRouter>
        <Admin />
      </MemoryRouter>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
