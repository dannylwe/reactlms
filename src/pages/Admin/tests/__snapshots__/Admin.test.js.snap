// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin should render without failing 1`] = `
Array [
  <nav
    className="navbar navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/dashboard"
    >
      Home
    </a>
    <a
      className="navbar-brand"
      href="/login"
      onClick={[Function]}
    >
      Logout
    </a>
  </nav>,
  <br />,
  <h2>
    Welcome Admin
  </h2>,
  <table>
    <tr>
      <th>
        Destination
      </th>
      <th>
        Order Nickname
      </th>
      <th>
        Status
      </th>
      <th>
        Pickup Location
      </th>
    </tr>
    loading......
  </table>,
]
`;
